R/spaces_left_parentheses_linter.R

```diff
@@ -3,61 +3,50 @@
 #' @export
 spaces_left_parentheses_linter <- function() {
   Linter(function(source_file) {
-    lapply(
-      ids_with_token(source_file, "'('"),
-      function(id) {
-
-        parsed <- source_file$parsed_content[id, ]
-
-        terminal_tokens_before <- with(
-          source_file$parsed_content,
-          token[line1 == parsed$line1 &
-                  col1 < parsed$col1 &
-                  terminal]
-        )
-        last_type <- tail(terminal_tokens_before, n = 1)
-
-        is_function <- length(last_type) %!=% 0L &&
-          (last_type %in% c("SYMBOL_FUNCTION_CALL", "FUNCTION", "'}'", "')'", "']'"))
-
-        if (!is_function) {
-
-          line <- source_file$lines[as.character(parsed$line1)]
-
-          before_operator <- substr(line, parsed$col1 - 1L, parsed$col1 - 1L)
-
-          non_space_before <- re_matches(before_operator, rex(non_space))
-          not_exception <- !(before_operator %in% c("!", ":", "[", "(", "^"))
-
-          # exception for unary - and unary +, #508
-          before_operator_idx <- with(
-            source_file$parsed_content,
-            col1 == parsed$col1 - 1L & col1 == col2
-          )
-          is_sibling_expr <- if (any(before_operator_idx)) {
-            with(
-              source_file$parsed_content,
-              token == "expr" & parent %in% parent[before_operator_idx]
-            )
-          } else {
-            rep(FALSE, nrow(source_file$parsed_content))
-          }
-          not_exception <- not_exception &&
-            !(before_operator %in% c("-", "+") &&
-              nrow(source_file$parsed_content[is_sibling_expr, ]) == 1L)
-
-          if (non_space_before && not_exception) {
-            Lint(
-              filename = source_file$filename,
-              line_number = parsed$line1,
-              column_number = parsed$col1,
-              type = "style",
-              message = "Place a space before left parenthesis, except in a function call.",
-              line = line
-            )
-          }
-        }
-
-      })
+
+    if (is.null(source_file$xml_parsed_content)) {
+      if (is.null(source_file$full_xml_parsed_content)) return(list())
```

Comment:
This if might be unreachable, either `xml_parsed_content` or `full_xml_parsed_content` is filled (unless maybe parsing fails, not sure about that).

Reply:
in fact it was this test that caused was triggering an error:

https://github.com/jimhester/lintr/blob/1aa166ef7aa3d85a667675847c277b83b4ff4908/tests/testthat/test-error.R#L3

Hence including this in 80a0139 -- maybe that same error doesn't show up on the codecov platform for some reason?